lib/dpl/providers/pages.rb

```diff
@@ -142,8 +153,20 @@ def git_status
         shell 'git status'
       end
 
+      def setup_deploy_key
+        path = "~/.dpl/deploy_key"
```

Comment:
I'd prefer it to be put into `ssh-agent` directly instead of storing in the file system.
In bash, you can do things like
```bash
ssh-agent '
  echo $ssh_key_var | ssh-add -
  ssh -T git@github.com  # check auth
  git ...
'
```
(IIRC the syntax)

Reply:
it's using the same strategy that other providers use and share, when using a per-build generated ssh key https://github.com/travis-ci/dpl/blob/master/lib/dpl/provider.rb#L332. i think it would be good to use the same approach for consistency, unless there's a strong reason?

Reply:
i agree there is an argument to be made that it would be good to not clutter the file system with these keys, for cases where people want to run this locally. although that is not the primary use case for dpl, it would make sense to optimize in that direction. however, i think if we considered that, we'd want to do it for all providers that use ssh keys, and stay with one approach.

Reply:
No specific reason other than security considerations, like a missing signal handler — I think it's safe to assume that in the event of SIGINT (Ctrl+C) or SIGTERM (kill) there's no robust cleanup set up that'd remove the key file.

I agree that it's probably okay'ish to keep it as is for consistency. As a future improvement, though, it'd be great if such files never hit the disk which could be implemented via a shared module that would be used across providers...

A few extra thoughts:
1) The script should set the folder privileges to 700 too.
2) IIUC semantically the best place for such folder would be under `$XDG_CACHE_HOME` (which is usually `~/.cache`), i.e. `~/.cache/dpl`. Or maybe even `$XDG_RUNTIME_DIR`. https://specifications.freedesktop.org/basedir-spec/latest/ar01s03.html

Reply:
>The script should set the folder privileges to 700 too.

sounds reasonable. i'll look into that.

> IIUC semantically the best place for such folder would be under $XDG_CACHE_HOME (which is usually ~/.cache), i.e. ~/.cache/dpl. Or maybe even $XDG_RUNTIME_DIR. https://specifications.freedesktop.org/basedir-spec/latest/ar01s03.html

hmmm, maybe. we used to use `./.dpl` for keeping these things (i.e. in the build working directory), which was a rather bad choice. i've just moved it to the home dir for the time being.